Malitsky-pock dies with assertion error

```
$ julia --project=scripts scripts/solve_qp.jl --instance_path /path/to/academictimetablesmall.mps.gz \
--method pdhg --output_dir /tmp/first_order_lp_solve --step_size_policy malitsky-pock \
--malitsky_pock_downscaling_factor=0.7 --malitsky_pock_interpolation_coefficient=0.4
```
yields
```
Instance: academictimetablesmall
runtime                  | residuals                  |  solution information      | relative residuals      |
#iter   #kkt     seconds | pr norm  du norm   gap     |  pr obj   pr norm  du norm | rel pr  rel du  rel gap |
0       2.5e+00  8.0e-01 | 6.1e+01  0.0e+00   7.0e+00 |  7.0e+00  0.0e+00  0.0e+00 | 3.0e-01 0.0e+00 8.8e-01 |
ERROR: LoadError: AssertionError: 0.0 <= target_radius < Inf
Stacktrace:
  [1] solve_bound_constrained_trust_region(center_point::Vector{Float64}, objective_vector::Vector{Float64}, variable_lower_bounds::Vector{Float64}, variable_upper_bounds::Vector{Float64}, norm_weights::Vector{Float64}, target_radius::Float64, solve_approximately::Bool)
    @ FirstOrderLp ~/FirstOrderLp/src/trust_region_utils.jl:88
  [2] bound_optimal_objective(problem::FirstOrderLp.QuadraticProgrammingProblem, primal_solution::Vector{Float64}, dual_solution::Vector{Float64}, primal_norm_weights::Vector{Float64}, dual_norm_weights::Vector{Float64}, distance_to_optimality::Float64, norm::FirstOrderLp.LocalizedDualityGapNorm; solve_approximately::Bool)
    @ FirstOrderLp ~/FirstOrderLp/src/trust_region_utils.jl:335
  [3] compute_localized_duality_gap
    @ ~/FirstOrderLp/src/saddle_point.jl:144 [inlined]
  [4] compute_localized_duality_gaps(problem::FirstOrderLp.QuadraticProgrammingProblem, current_primal_solution::Vector{Float64}, current_dual_solution::Vector{Float64}, avg_primal_solution::Vector{Float64}, avg_dual_solution::Vector{Float64}, primal_norm_params::Vector{Float64}, dual_norm_params::Vector{Float64}, last_restart_info::FirstOrderLp.RestartInfo, use_approximate_localized_duality_gap::Bool)
    @ FirstOrderLp ~/FirstOrderLp/src/saddle_point.jl:411
  [5] run_restart_scheme(problem::FirstOrderLp.QuadraticProgrammingProblem, solution_weighted_avg::FirstOrderLp.SolutionWeightedAverage, current_primal_solution::Vector{Float64}, current_dual_solution::Vector{Float64}, last_restart_info::FirstOrderLp.RestartInfo, iterations_completed::Int64, primal_norm_params::Vector{Float64}, dual_norm_params::Vector{Float64}, primal_weight::Float64, verbosity::Int64, restart_params::FirstOrderLp.RestartParameters)
    @ FirstOrderLp ~/FirstOrderLp/src/saddle_point.jl:683
  [6] optimize(params::FirstOrderLp.PdhgParameters, original_problem::FirstOrderLp.QuadraticProgrammingProblem)
    @ FirstOrderLp ~/FirstOrderLp/src/primal_dual_hybrid_gradient.jl:968
  [7] macro expansion
    @ ~/FirstOrderLp/scripts/solve_qp.jl:110 [inlined]
  [8] macro expansion
    @ ./timing.jl:287 [inlined]
  [9] (::var"#inner_solve#13"{FirstOrderLp.PdhgParameters, String, String, Bool, Bool, String})()
    @ Main ~/FirstOrderLp/scripts/solve_qp.jl:109
 [10] solve_instance_and_output(parameters::FirstOrderLp.PdhgParameters, output_dir::String, instance_path::String, redirect_stdio::Bool, transform_bounds::Bool, fixed_format_input::Bool)
    @ Main ~/FirstOrderLp/scripts/solve_qp.jl:158
 [11] main()
    @ Main ~/FirstOrderLp/scripts/solve_qp.jl:591
 [12] top-level scope
    @ ~/FirstOrderLp/scripts/solve_qp.jl:601
```

This error happens with high frequency in our experiments, above is just one example. `academictimetablesmall.mps.gz` is generated by the instructions at https://github.com/google-research/FirstOrderLp.jl/tree/main/benchmarking#filtered-and-preprocessed-miplib-2017-collection-dataset.

--- a/src/primal_dual_hybrid_gradient.jl
+++ b/src/primal_dual_hybrid_gradient.jl
@@ -614,8 +614,9 @@ function take_step(
         next_dual_product,
       )
       done = true
+    else
+      step_size *= step_params.downscaling_factor
     end
-    step_size *= step_params.downscaling_factor
   end
   if iter == max_iter && !done
     solver_state.numerical_error = true

